### PR TITLE
Smoother publishing process with preversion

### PR DESCRIPTION
**Pull request by josojo**
This PR includes the npm command `preversion`, in order to improve the publishing process.
It ensures that the newest contracts are build and that new newest addresses from the networks.json are being published.

### CODE REVIEW

**Comment by nlordell** [comment]
I was under the impression that `prepack` gets called as well making `preversion` is redundant - I may be wrong though, it is worth checking. It could be that we just want to `rm -rf build` in `prepack` script.

**Review by nlordell — COMMENTED** [review]
I was under the impression that prepack gets called as well making preversion is redundant - I may be wrong though, it is worth checking. It could be that we just want to rm -rf build in prepack script.

**Comment by nlordell** [comment]
According to the npm docs, it looks like `prepack` is run everytime before an `npm pack` or `npm publish`. So you could just do `npm version && npm publish` and the contracts will be built. It might make sense to remove the build folder though and do a `networks-reset` from there, to ensure that the build is pristine and does not include dev networks in it.

**Review by nlordell — APPROVED** [review]
Looks great!

File: package.json [hunk 1]
```diff
@@ -13,11 +13,11 @@
     "coverage": "solidity-coverage",
     "networks-extract": "CONF_FILE=$(pwd)'/migration_conf.js'  node node_modules/@gnosis.pm/util-contracts/src/extract_network_info.js",
     "networks-inject": "CONF_FILE=$(pwd)'/migration_conf.js'  node node_modules/@gnosis.pm/util-contracts/src/inject_network_info.js",
-    "networks-reset": "mkdir -p build/contracts && npx truffle networks --clean && npm run networks-inject",
+    "networks-reset": "mkdir -p build/contracts && truffle networks --clean && npm run networks-inject",
```
Comment (fleupold) [inline package.json]: Why this change? It will require a global truffle

**Comment by josojo** [comment]
I have overlooked the `prepack`. Sorry.
In the past, we used `preversion`, as this runs the cleaning procedure before the new commit with the new version number. But since all the cleaning procedure will only clean files, which are in .gitignore, your method is actually better. 
Cool!

File: package.json (hunk 1 above)
Comment (nlordell) [reply package.json]: No, npm scripts are run with `node_modules/.bin/` automatically added to the `PATH`.

Comment (josojo) [reply package.json]: No, it will use the truffle specified in the dependencies.

I could not find a better source than this one:
https://dev.to/nebrius/comment/a9l9

But Nick said this in the past as well :)

Comment (nlordell) [reply package.json]: Here :) https://docs.npmjs.com/cli/run-script
> In addition to the shell’s pre-existing PATH, npm run adds node_modules/.bin to the PATH provided to scripts.

Comment (nlordell) [reply package.json]: Or here now https://yarnpkg.com/lang/en/docs/cli/run/
> [script] can also be any locally installed executable that is inside node_modules/.bin/
